cohortextractor/main.py

```diff
@@ -54,15 +69,18 @@ def extract(cohort, backend):
             yield dict(row)
 
 
-def write_output(results, output_file):
+def write_output(results, output_file, dummy_data_file=None):
```

Comment:
I don't think that we gain anything from having the two different ways of producing output files in this same function. It would be clearer to keep the conditionals all in one place (in `main()`). Something like this, in pseudo-code:

```
cohort = load_cohort()

if dummy_data:
    validate(dummy_data, cohort)
    copy(dummy_data, output_file)
else:
    results = extract(cohort)
    write(results, output_file)
```

Reply:
I was trying to keep `main` doing fewer things, but sure, this is fine too